lib/iban_calculator.rb

```diff
@@ -36,7 +36,7 @@ def self.execute(method, options = {})
     client = Savon.client(wsdl: config.url, logger: config.logger)
     client.call(method, message: options).tap do |response|
       status = response.body[:"#{method}_response"][:return][:result]
-      fail(ServiceError, status) unless status == 'passed'
+      fail(ServiceError, status) unless response.body[:"#{method}_response"][:return_code]
```

Comment:
Does this really always work? I am not sure, but I think to remember that the response keys not always had the same format "#{method}_response". But I might be wrong.